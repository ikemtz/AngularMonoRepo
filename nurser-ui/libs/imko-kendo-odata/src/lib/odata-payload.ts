export interface ODataPayload<T> {
  '@odata.context'?: string;
  '@odata.count'?: number;
  value: T[];
}
