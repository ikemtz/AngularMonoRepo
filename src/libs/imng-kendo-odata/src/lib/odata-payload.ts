export interface ODataPayload<T> {
  '@odata.context'?: string;
  '@odata.count'?: number;
  value: T[];
}

export function createODataPayload<T>(resultSet: T[]): ODataPayload<T> {
  return {
    ['@odata.count']: resultSet?.length,
    value: resultSet,
  };
}
