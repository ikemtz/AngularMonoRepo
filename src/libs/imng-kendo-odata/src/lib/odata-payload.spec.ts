import { createODataPayload } from './odata-payload';

describe('createODataPayload', () => {

  it('should work', () => {
    const resultSet = [1, 2, 3, 4];
    const result = createODataPayload(resultSet);
    expect(result).toMatchSnapshot();
  });
});
