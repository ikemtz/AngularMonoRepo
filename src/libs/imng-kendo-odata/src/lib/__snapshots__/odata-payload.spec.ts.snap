// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createODataPayload should work 1`] = `
Object {
  "@odata.count": 4,
  "value": Array [
    1,
    2,
    3,
    4,
  ],
}
`;
